src/metrics.ts

```diff
@@ -184,30 +179,34 @@ export const MetricsProcessor = (
 
   const _send = (): void => {
     if (closed) {
+      log.debug('SDK has been closed before metrics can be sent');
+      return;
+    }
+
+    if (data.size == 0) {
+      log.debug('No metrics to send in this interval');
       return;
     }
 
     const metrics: Metrics = _summarize();
 
-    if (metrics && metrics.metricsData.length && metrics.targetData.length) {
```

Comment:
We can safely remove this check, because the earlier `data` check now works correctly.